pkg/apis/core/validation/validation.go

```diff
@@ -2647,21 +2691,31 @@ func validateEphemeralContainers(ephemeralContainers []core.EphemeralContainer,
 		}
 
 		// Ephemeral Containers should not be relied upon for fundamental pod services, so fields such as
-		// Lifecycle, probes, resources and ports should be disallowed. This is implemented as a whitelist
-		// so that new fields will be given consideration prior to inclusion in Ephemeral Containers.
-		specType, specValue := reflect.TypeOf(ec.EphemeralContainerCommon), reflect.ValueOf(ec.EphemeralContainerCommon)
-		for i := 0; i < specType.NumField(); i++ {
-			f := specType.Field(i)
-			if allowedEphemeralContainerFields[f.Name] {
-				continue
-			}
+		// Lifecycle, probes, resources and ports should be disallowed. This is implemented as a list
+		// of allowed fields so that new fields will be given consideration prior to inclusion in Ephemeral Containers.
+		allErrs = append(allErrs, validateFieldAllowList(ec.EphemeralContainerCommon, allowedEphemeralContainerFields, "cannot be set for an Ephemeral Container", idxPath)...)
```

Comment:
Ephemeral Container -> Ephemeral Volume

Reply:
No here. This really is the code for `ec.EphemeralContainerCommon`, I just touched it and moved code into  `validateFieldAllowList` because I wanted to use the same code also for validating allowed fields in `validatePersistentVolumeClaimTemplateObjectMeta`

Reply:
Oh, sorry about the noise